### PR TITLE
Add support to OTEL_DOTNET_AUTO_LOGS_ENABLED

### LINKED ISSUE #1479: Disable Logs globally
Currently we have `OTEL_DOTNET_AUTO_LOAD_*_AT_STARTUP` for other signals. We miss similar functionality for logs.

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Fixes #1479

## What

Add support to `OTEL_DOTNET_AUTO_LOGS_ENABLED`.
Handles `OTEL_DOTNET_AUTO_TRACES_ENABLED` in native code.
Small refactor in the road to #1482.

## Tests

CI

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] Documentation is updated.
- [x] New features are covered by tests.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ This component adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.h
 ### Added
 
 - Add support for .NET 7.
+- Add support for `OTEL_DOTNET_AUTO_TRACES_ENABLED`.
```
Comment (pellared) [inline CHANGELOG.md:37]: How about adding info in `### Fixed` that the ``OTEL_DOTNET_AUTO_*_ENABLED` are also respected by bytecode instrumentations?

**Review by pellared — APPROVED** [review]
We should add a SmokeTests for `OTEL_DOTNET_AUTO_*_ENABLED=false` (can be as a separate PR). It should be very easy (based on https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1613).

**Review by pellared — APPROVED** [review]

File: src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.cpp [hunk 2]
```diff
@@ -13,37 +13,25 @@ using json = nlohmann::json;
...
-    const std::vector<WSTRING>& disabledTraceIntegrationNames,
-    const std::vector<WSTRING>& enabledLogIntegrationNames,
-    const std::vector<WSTRING>& disabledLogIntegrationNames) {
+    const LoadIntegrationConfiguration& configuration) {
     for (const WSTRING& filePath : GetEnvironmentValues(environment::integrations_path, ENV_VAR_PATH_SEPARATOR))
     {
         Logger::Debug("Loading integrations from file: ", filePath);
-      LoadIntegrationsFromFile(
-          filePath, integrationMethods, enabledTraceIntegrationNames, disabledTraceIntegrationNames, enabledLogIntegrationNames, disabledLogIntegrationNames);
+      LoadIntegrationsFromFile(filePath, integrationMethods, configuration);
```
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.cpp:20]:
```suggestion
        LoadIntegrationsFromFile(filePath, integrationMethods, configuration);
```

File: src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.cpp [hunk 3]
```diff
@@ -86,12 +71,7 @@ void LoadIntegrationsFromStream(
 
         for (const auto& el : j)
         {
-          IntegrationFromJson(el,
-                              integrationMethods,
-                              enabledTraceIntegrationNames,
-                              disabledTraceIntegrationNames,
-                              enabledLogIntegrationNames,
-                            disabledLogIntegrationNames);
+          IntegrationFromJson(el, integrationMethods, configuration);
```
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.cpp:74]:
```suggestion
            IntegrationFromJson(el, integrationMethods, configuration);
```

File: src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.cpp [hunk 4]
```diff
@@ -181,14 +158,24 @@ namespace
 
         if (type == WStr("Trace"))
         {
-            if (!InstrumentationEnabled(name, enabledTraceIntegrationNames, disabledTraceIntegrationNames)) 
+            if (!configuration.traces_enabled)
+            {
+                  return;
```
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.cpp:163]:
```suggestion
                return;
```

**Review by pjanotti — APPROVED** [review]
LGTM, just some nits about indentation.

**Comment by Kielek** [comment]
@pjanotti, I do not want to fix indentation suggestion. We have separate task to introduce common formating standard. Without this enforcement any changes in cop files are nightmare.

**Pull request merged**